### PR TITLE
Hacky struct stats fix

### PR DESCRIPTION
**Pull request by Tom-Newton**
# Description
The description of the main changes of your pull request

# Related Issue(s)

# Documentation

### CODE REVIEW

File: rust/src/action.rs [hunk 1]
```diff
@@ -288,19 +288,36 @@ impl Add {
...
+            }
+            Err(e) => {
+                log::error!("Could not read parquet stats {:?} {e}", self.stats_parsed);
+            }
+            Ok(_) => {
+                log::debug!("No parquet stats")
+            }
+        }
+        let stats = self.get_json_stats();
+        return stats;
```
Comment (wjones127) [inline rust/src/action.rs:305]:
```suggestion
        match self.get_stats_parsed() {
            Ok(Some(stats)) => Ok(Some(stats)),
            Ok(None) => self.get_json_stats(),
            Err(e) => {
                log::error!("Could not read parquet stats {:?} {e}", self.stats_parsed);
            }
        }
```

File: rust/src/action.rs [hunk 2]
```diff
@@ -364,6 +381,22 @@ impl Add {
...
+fn parquet_field_to_json(field: Field) -> serde_json::Value {
+    match field {
+        Field::Bool(val) => json!(val),
+        Field::Byte(val) => json!(val),
+        Field::Short(val) => json!(val),
+        Field::Int(val) => json!(val),
+        Field::Long(val) => json!(val),
+        Field::Float(val) => json!(val),
+        Field::Double(val) => json!(val),
+        // Field::Decimal(val) => json!(val),
```
Comment (wjones127) [inline rust/src/action.rs:393]: I believe decimal needs to be converted to a string.

File: rust/src/action.rs [hunk 3]
```diff
@@ -364,6 +381,22 @@ impl Add {
...
+        Field::Str(val) => json!(val),
+        // Field::Bytes(val) => json!(val),
```
Comment (wjones127) [inline rust/src/action.rs:395]: Bytes needed to be encoded in an escaped format. `std::ascii::escape_default` might be a good candidate for encoding that.

**Review by wjones127 — COMMENTED** [review]
Thanks for creating this! I've left a few comments.

**Comment by wjones127** [comment]
I think these changes are reasonable FYI. Feel free to make a PR into delta-io/delta-rs once you've addressed the above comments and I think we should be nearly ready to merge.

**Comment by Tom-Newton** [comment]
Thanks for taking a look @wjones127. I've been trying to work out the conversions for the types I skipped initially. I'm a bit unsure though what the requirements really are. I feel like the output should really be identical to what we would get if the same stats were parsed from json? 

So for example Decimal types appear to be stored as json Number types not String.

I did also discover https://github.com/apache/arrow-rs/blob/9f7b6004d365b0c0bac8e30170b49bdd66cc7df0/parquet/src/record/api.rs#L653-L701 which appears initially to do exactly what we want. Howver it does not parse the parquet to the same json form we would get from json stats. Some discrepancies:
1. It omits milliseconds from timestmamps
2. It puts timezones on date and timestamp types
3. It creates Stings instead of Numbers for decimals 
4. Struct columns need to be parsed to nested `ColumnStatValue`s not just json Objects.

**Comment by Tom-Newton** [comment]
This was merged upstream

**Pull request closed without merging**